src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,126 @@
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+
+
+@Patch(
+    name = "Change link sharing domain",
+    description = "Replaces the domain name of Twitter links when sharing them.",
+    compatiblePackages = [CompatiblePackage("com.twitter.android")]
+)
+@Suppress("unused")
+object ChangeLinkSharingDomainPatch : BytecodePatch(
+    setOf(LinkBuilderFingerprint, LinkResourceGetterFingerprint)
+) {
+    private var domainName by stringPatchOption(
+        key = "domainName",
+        default = "fxtwitter.com",
+        title = "Domain name",
+        description = "The domain to use when sharing links.",
+        required = true
+    )
+
+    override fun execute(context: BytecodeContext) {
+        val linkBuilderResult = LinkBuilderFingerprint.result ?: throw LinkBuilderFingerprint.exception
+
+        // region Copy link button.
+        val buildShareLinkMethod = linkBuilderResult.mutableMethod.apply {
+            val stringIndex = linkBuilderResult.scanResult.stringsScanResult!!.matches
+                .first().index
+
+            val targetRegister = getInstruction<OneRegisterInstruction>(stringIndex).registerA
+            replaceInstruction(
+                stringIndex,
+                "const-string v$targetRegister, \"https://$domainName/%1\$s/status/%2\$d\""
+            )
+        }
+
+
+        // endregion
+
+        // Used in the Share via... dialog.
+        LinkResourceGetterFingerprint.result?.apply {
+            // Remove the original call to the method that uses the resources template to build the link.
+            val originalMethodIndex = mutableMethod.indexOfFirstInstruction {
+                opcode == Opcode.INVOKE_VIRTUAL &&
+                        getReference<MethodReference>()?.definingClass == "Landroid/content/res/Resources;"
+            }
+            if (originalMethodIndex == -1) throw PatchException("Could not find originalMethodIndex")
+
+            val shareLinkRegister =
+                mutableMethod.getInstruction<OneRegisterInstruction>(originalMethodIndex + 1).registerA
+
+            mutableMethod.apply {
+                removeInstructions(originalMethodIndex, 2)
+
+                // Remove the instruction that uses the shareLinkRegister as an array reference to prevent an error.
+                removeInstructions(originalMethodIndex - 2, 1)
+            }
+
+
+            // region Get the nickname of the user that posted the tweet. This is used to build the link.
+            val getNicknameIndex = mutableMethod.indexOfFirstInstruction {
+                opcode == Opcode.INVOKE_VIRTUAL &&
+                        getReference<MethodReference>().toString().endsWith("Ljava/lang/String;")
+            }
+            if (getNicknameIndex == -1) throw PatchException("Could not find getNicknameIndex")
+
+            val sourceNicknameRegister =
+                this.mutableMethod.getInstruction<OneRegisterInstruction>(getNicknameIndex + 1).registerA
```

Comment:
What is `getNicknameIndex + 1`?

Reply:
I thought it was kinda obvious. Method gets a nickname and saves it to a register in the next instruction. Will try to document it ig